client/store/product.js

```diff
@@ -0,0 +1,139 @@
+import axios from 'axios'
+
+const GOT_PRODUCTS = 'GOT_PRODUCTS'
+const GOT_PRODUCT_INFO = 'GOT_PRODUCT_INFO'
+const GOT_FILTERED_PRODUCTS = 'GET_PRODUCTS'
+const GOT_NEW_PRODUCT = 'GOT_NEW_PRODUCT'
+const DELETE_PRODUCT = 'DELETE_PRODUCT'
+const GOT_UPDATED_PRODUCT = 'GOT_UPDATED_PRODUCT'
+
+const gotProducts = products => ({type: GOT_PRODUCTS, products})
+const gotProductInfo = productInfo => ({type: GOT_PRODUCT_INFO, productInfo})
+const gotNewProduct = newProduct => ({type: GOT_NEW_PRODUCT, newProduct})
+const deleteProduct = removedProductId => ({
+  type: DELETE_PRODUCT,
+  removedProductId
+})
+const gotUpdatedProduct = updatedProduct => ({
+  type: GOT_UPDATED_PRODUCT,
+  updatedProduct
+})
+const gotFilteredProducts = filteredProducts => ({
+  type: GOT_FILTERED_PRODUCTS,
+  filteredProducts
+})
+
+// const fetchingProducts = () => ( { type: PRODUCTS_REQUEST} )
+// const fetchingProductInfo = () => ( { type: PRODUCTS_INFO_REQUEST } )
+// const productError = error => ( { type: PRODUCT_ERROR, error } )
+//May not need the above three action creators, depending on where we decide to handle errors
+//and/or loading bar
+
+export const fetchProducts = () => {
+  return async dispatch => {
+    try {
+      const {data: products} = await axios.get('/api/products')
+      dispatch(gotProducts(products))
+    } catch (error) {
+      console.error(error)
+    }
+  }
+}
+
+export const fetchAndFilterProducts = productCategoryId => {
+  return async dispatch => {
+    try {
+      const {data: products} = await axios.get('/api/products')
+      const filteredProducts = products.filter(
+        product => product.categoryId === productCategoryId
+      )
```

Comment:
Instead of fetching all products and filtering on the client side, implement query string filtering in the backend (which in turn should become a `where` clause to the database).